### PR TITLE
chore: cleanup and add sample staking delegation to demo

### PR DESCRIPTION
**Pull request by damiannolan**
- Various cleanup
- Added a new cmd to perform ICA staking delegation over ibc
- Updated README.md demo to include staking delegation

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
utACK (did not do a close review)

File: proto/intertx/tx.proto [hunk 1]
```diff
@@ -43,3 +47,20 @@ message MsgSend {
...
+
+// MsgDelegate builds and sends a stakingtypes.MsgDelegate to the interchian account associated with the provided owner and connection pair
+// The interchain account on the associated host chain will execute the stakingtypes.MsgDelegate
+message MsgDelegate {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string   interchain_account       = 1 [(gogoproto.moretags) = "yaml:\"interchain_account\""];
+  bytes    owner                    = 2 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/types.AccAddress"];
+  string   validator_address        = 3 [(gogoproto.moretags) = "yaml:\"interchain_account\""];
```
Comment (seantking) [inline proto/intertx/tx.proto:59]:
```suggestion
  string   validator_address        = 3 [(gogoproto.moretags) = "yaml:\"validator_address\""];
```

File: x/inter-tx/client/cli/flags.go [hunk 2]
```diff
@@ -5,16 +5,18 @@ import (
 )
 
 const (
-	FlagConnectionId             = "connection-id"
-	FlagCounterpartyConnectionId = "counterparty-connection-id"
+	// The connection end identifier on the controller chain
+	FlagConnectionID = "connection-id"
+	// The connection end identifier on the host chain
+	FlagCounterpartyConnectionID = "counterparty-connection-id"
```
Comment (seantking) [inline x/inter-tx/client/cli/flags.go:11]: should we just remove this and refactor to get the counterparty connection ID from the connection ID?

File: x/inter-tx/client/cli/tx.go [hunk 3]
```diff
@@ -74,36 +66,68 @@ func getRegisterAccountCmd() *cobra.Command {
 
 func getSendTxCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:  "send [interchain_account_address] [to_address] [amount] --connection-id --counterparty-connection-id",
+		Use:  "send [interchain_account_address] [to_address] [amount]",
```
Comment (seantking) [inline x/inter-tx/client/cli/tx.go:69]: same question here re: flags

File: x/inter-tx/types/query.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package types
+
+// NewQueryInterchainAccountRequest creates and returns a new QueryInterchainAccountFromAddressRequest
+func NewQueryInterchainAccountRequest(owner, connectionID, counterpartyConnectionID string) *QueryInterchainAccountFromAddressRequest {
+	return &QueryInterchainAccountFromAddressRequest{
+		Owner:                    owner,
+		ConnectionId:             connectionID,
+		CounterpartyConnectionId: counterpartyConnectionID,
```
Comment (seantking) [inline x/inter-tx/types/query.go:8]: We can probs remove the counterparty connection here also (doesn't need to be in this PR) just food for thought.

**Comment by seantking** [comment]
This is cool. Group module next! :D

File: proto/intertx/tx.proto (hunk 1 above)
Comment (damiannolan) [reply proto/intertx/tx.proto:59]: Good spot!

File: x/inter-tx/client/cli/flags.go (hunk 2 above)
Comment (damiannolan) [reply x/inter-tx/client/cli/flags.go:11]: I can look at doing that in a follow up PR for sure.

File: x/inter-tx/client/cli/tx.go [hunk 5]
```diff
@@ -28,31 +23,28 @@ func GetTxCmd() *cobra.Command {
...
+		getDelegateTxCmd(),
 	)
 
 	return cmd
 }
 
 func getRegisterAccountCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "register --connection-id --counterparty-connection-id",
+		Use: "register",
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply x/inter-tx/client/cli/tx.go:37]: The flags are not required in the `Use` string to actually be present. The flags are still required by:

```
cmd.Flags().AddFlagSet(fsConnectionPair)

_ = cmd.MarkFlagRequired(FlagConnectionID)
_ = cmd.MarkFlagRequired(FlagCounterpartyConnectionID)

flags.AddTxFlagsToCmd(cmd)
```

File: x/inter-tx/client/cli/tx.go (hunk 3 above)
Comment (damiannolan) [reply x/inter-tx/client/cli/tx.go:69]: See above!

File: x/inter-tx/types/query.go (hunk 4 above)
Comment (damiannolan) [reply x/inter-tx/types/query.go:8]: Agreed

**Comment by damiannolan** [comment]
> This is cool. Group module next! :D

Yeah, for sure! A more fully fledged and real world demo would be great, but for now at least we are exercising critical code paths!

File: x/inter-tx/client/cli/tx.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (seantking) [reply x/inter-tx/client/cli/tx.go:37]: I think the use string is just a guide for using the command, so we should probably leave the flags in

**Comment by seantking** [comment]
> > This is cool. Group module next! :D
> 
> Yeah, for sure! A more fully fledged and real world demo would be great, but for now at least we are exercising critical code paths!

group and gov module actually aren't ready yet so not much we can do yet. I think focusing on creating/using LPs would be a nice one too

File: x/inter-tx/client/cli/tx.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (damiannolan) [reply x/inter-tx/client/cli/tx.go:37]: But `[flags]` is implicitly appended to the usage string. If you run:

```
icad tx intertx register

Error: required flag(s) "connection-id", "counterparty-connection-id" not set
Usage:
  icad tx intertx register [flags]
``` 

If you opt to leave them in it appears as below, which doesn't look quiet right.

```
icad tx intertx register

Error: required flag(s) "connection-id", "counterparty-connection-id" not set
Usage:
  icad tx intertx register --connection-id --counterparty-connection-id [flags]
``` 

Are you sure you want to leave them in?

[the remark this answers is not in the document]
Comment (seantking) [reply x/inter-tx/client/cli/tx.go:37]: ahh nice, I learn something new. It's good to remove them then!

**Review by seantking — APPROVED** [review]

File: x/inter-tx/client/cli/tx.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (damiannolan) [reply x/inter-tx/client/cli/tx.go:37]: Sweet, I'll merge this then 👍

**Pull request merged**